### PR TITLE
treeForTranslations hook

### PR DESCRIPTION
**Pull request by jasonmit**
The goal is summed up here https://github.com/ember-intl/ember-intl/issues/401#issuecomment-316629409

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -19,114 +18,205 @@ const TranslationReducer = require('./lib/broccoli/translation-reducer');
...
+    addons.forEach(addon => this.processAddon(addon, nodes));
+  },
+
+  /*
+  * @method processAddon
+  * @param {object} addon model
+  * @param {array} array of broccoli node/tree/directories
+  * @private
+  */
+  processAddon(addon, nodes) {
```
Comment (jasonmit) [inline index.js]: Ripped from ember-cli-fastboot.  They are the only addon I know that introduces an addon hook and it looked like a good implementation for what I need.

File: index.js [hunk 2]
```diff
@@ -19,124 +18,218 @@ const TranslationReducer = require('./lib/broccoli/translation-reducer');
...
+    /* walk through each addon and grab its translation tree */
+    let addonTranslationPath = path.join(addon.root, this.translationPathForAddon(addon));
+    let _translationTree;
+
+    if (existsSync(addonTranslationPath)) {
+      /* verify these are not setting up watchers */
```
Comment (jasonmit) [inline index.js]: I probablly want `UnwatchedDir` here

**Comment by vlascik** [comment]
Nice, could use a mention in docs though? People might not find out about this otherwise.

**Comment by jasonmit** [comment]
@vlascik it's on master, which is not versioned.  It will be documented when it is released and the APIs are marked "public".

**Pull request merged**